Lombiq.Tests.UI/Services/UITestExecutionSession.cs

```diff
@@ -428,6 +425,27 @@ private void CaptureMarkupValidationResults(Exception ex, string debugInformatio
         }
     }
 
+    private void LogRetry(int retryCount)
+    {
+        _testOutputHelper.WriteLineTimestampedAndDebug(
+            "The test was attempted {0} time(s). {1} more attempt(s) will be made after waiting {2}.",
+            retryCount + 1,
+            _configuration.MaxRetryCount - retryCount,
+            _configuration.RetryInterval);
+
+        if (_configuration.ExtendGitHubActionsOutput &&
+            _configuration.GitHubActionsOutputConfiguration.EnableTestRetryWarningAnnotations &&
+            GitHubHelper.IsGitHubEnvironment)
+        {
+            new GitHubAnnotationWriter(_testOutputHelper).Annotate(
```

Comment:
This is how this looks like: https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions/actions/runs/3550298479